sql/planner/hash_set.go

```diff
@@ -39,10 +39,14 @@ func (s documentHashSet) generateKey(d document.Document) (uint64, error) {
 			return err
 		}
 
+		buf = buf[0:]
```

Comment:
This seems redundant, unless `var buf []byte` is declared outside of the `Iterate` function.

Reply:
Also, should be `buf[:0]` to reset slice length.